modules/andes-core/client/src/main/java/org/wso2/andes/client/BasicMessageConsumer.java

```diff
@@ -619,7 +619,9 @@ public void close(boolean sendClose) throws JMSException
                         // no point otherwise as the connection will be gone
                         if (!_session.isClosed() || _session.isClosing())
                         {
-                            _session._dispatcher.rejectPending(this);
+                            if(null != _session._dispatcher) {
```

Comment:
Can it lead to any errors if we do not `rejectPending` since the dispatcher is null?